src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/ConstructorOnTypeBuilderInstantiation.cs

```diff
@@ -58,7 +58,7 @@ public override object Invoke(object? obj, BindingFlags invokeAttr, Binder? bind
         public override CallingConventions CallingConvention => _ctor.CallingConvention;
         public override Type[] GetGenericArguments() { return _ctor.GetGenericArguments(); }
         public override bool IsGenericMethodDefinition => false;
-        public override bool ContainsGenericParameters => _type.ContainsGenericParameters;
+        public override bool ContainsGenericParameters => _ctor.ContainsGenericParameters;
```

Comment:
What the reasoning behind this change? The original implementation looks correct to me.

Reply:
> What the reasoning behind this change? 

The reason for this change is we don't want to load the original ctor when the ctor is instance of `ConstructorOnTypeBuilderInstantiation`, but we cannot check that outside of CoreLib because it is internal type, instead checking `!methodBase.ContainsGenericParameters` on method or ctor

https://github.com/dotnet/runtime/blob/be5694d8e8494f533bfe5d48d987f5e050df80d2/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs#L609-L613

For ctors for example populated like:
 `TypeBuilder.GetConstructor(returnType, typeof(List<>).GetConstructor([typeof(int)]))`
Here the `_type.ContainsGenericParameters` almost always will be false, but the `_ctor.ContainsGenericParameters` always will be true


> The original implementation looks correct to me.

I would say this is not (completely) wrong, probably we should check both for completeness, same as in `MethodOnTypeBuilderInstantiation`
https://github.com/dotnet/runtime/blob/be5694d8e8494f533bfe5d48d987f5e050df80d2/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/MethodOnTypeBuilderInstantiation.cs#L82-L85

Reply:
We seem to be missing test coverage for `ContainsGenericParameters` property. The following program prints true + false before this PR, but it is pritns false + false after this PR; but all tests are passing. I think that the behavior before this PR is the correct one.
```
using System.Reflection;
using System.Reflection.Emit;

AssemblyBuilder ab = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("test"), AssemblyBuilderAccess.RunAndCollect);
ModuleBuilder mb = ab.DefineDynamicModule("test");
TypeBuilder tb = mb.DefineType("test", TypeAttributes.Public);
var typeParameterBuilder = tb.DefineGenericParameters("T");
var constructorBuilder = tb.DefineDefaultConstructor(MethodAttributes.Public);

var instantiatedTypeBuilder1 = tb.MakeGenericType(typeof(List<>).GetGenericArguments()[0]);
Console.WriteLine(TypeBuilder.GetConstructor(instantiatedTypeBuilder1, constructorBuilder).ContainsGenericParameters);

var instantiatedTypeBuilder2 = tb.MakeGenericType(typeof(int));
Console.WriteLine(TypeBuilder.GetConstructor(instantiatedTypeBuilder2, constructorBuilder).ContainsGenericParameters);
```

https://github.com/dotnet/runtime/blob/be5694d8e8494f533bfe5d48d987f5e050df80d2/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/MethodOnTypeBuilderInstantiation.cs#L82 does not look right to me. I think it should be just ` if (_type.ContainsGenericParameters)`.

Reply:
> the ctor is instance of ConstructorOnTypeBuilderInstantiation (because GetMemberWithSameMetadataDefinitionAs is not implemeneted),

Is it possible to implement it to address the issue?

Reply:
> > the ctor is instance of ConstructorOnTypeBuilderInstantiation (because GetMemberWithSameMetadataDefinitionAs is not implemeneted),
> 
> Is it possible to implement it to address the issue?

Sorry I mistaken, its calling `GetMemberWithSameMetadataDefinitionAs(methodBase)` on `declaringType.GetGenericTypeDefinition()` not on `ConstructorOnTypeBuilderInstantiation` which means it will be called on the open generic runtime type, and it could not find the matching member when the member includes non runtime type as generic parameter. 

Though I think I can avoid using `methodBase.ContainsGenericParameters` by checking the generic parameters.

> We seem to be missing test coverage for ContainsGenericParameters property. 

I will add tests